Google.Api.Gax.Grpc/ClientBuilder.cs

```diff
@@ -0,0 +1,205 @@
+﻿/*
+ * Copyright 2019 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Apis.Auth.OAuth2;
+using Grpc.Auth;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// Base class for API-specific builders.
+    /// </summary>
+    /// <typeparam name="TClient">The type of client created by this builder.</typeparam>
+    public abstract class ClientBuilder<TClient>
+    {
+        /// <summary>
+        /// The endpoint to connect to, or null to use the default endpoint.
+        /// </summary>
+        public ServiceEndpoint Endpoint { get; set; }
+
+        /// <summary>
+        /// The scopes to use, or null to use the default scopes.
+        /// </summary>
+        public IReadOnlyList<string> Scopes { get; set; }
+
+        /// <summary>
+        /// The channel credentials to use, or null if credentials are being provided in a different way.
+        /// </summary>
+        public ChannelCredentials ChannelCredentials { get; set; }
+
+        /// <summary>
+        /// The path to the credentials file to use, or null if credentials are being provided in a different way.
+        /// </summary>
+        public string CredentialsPath { get; set; }
+
+        /// <summary>
+        /// The credentials to use as a JSON string, or null if credentials are being provided in a different way.
+        /// </summary>
+        public string JsonCredentials { get; set; }
+
+        /// <summary>
+        /// The token access method to use, or null if credentials are being provided in a different way.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// To use a GoogleCredential for credentials, call this method using a method group conversion, e.g.
```

Comment:
nit: This part of the comment is outdated.

Reply:
Done.